src/ERC20.sol

```diff
@@ -35,10 +35,10 @@ contract ERC20 is IERC20 {
 
     uint256 public override totalSupply;
 
-    string public name;
-    string public symbol;
+    string public override name;
```

Comment:
Now that we have split the token deployment from the gateway, I think it's worth having proper methods for these for clarity. Would the increased bytecode size have a significant gas impact?

Otherwise, can you add to the doc above that allowance etc. implement the IERC20 interface because of how variables are referred to via public methods?

Reply:
"Otherwise, can you add to the doc above that allowance etc. implement the IERC20 interface because of how variables are referred to via public methods?" I don't think that's necessary since it's a common knowledge of solidity.